enhancements/machine-api/cluster-api-integration.md

```diff
@@ -0,0 +1,268 @@
+
+### Non-Goals
+
+- Deprecate or remove any existing APIs.
+- Stop providing support for Machine API in near future.
+- Provide any automated integration or migration between MAPI and CAPI resources.
+
+## Proposal
+
+This proposal is about introducing Cluster API alongside Machine API as a technical preview in OpenShift clusters. Cluster API on OpenShift has the potential to unlock new infrastructure providers and community engagement for our users.
+During the technical preview we will gather feedback on its usefulness as well as evaluate the feasibility of using Cluster API as a primary infrastructure resource API for OpenShift.
+
+### User Stories
+
+#### Story 1
+
+As an OpenShift developer, I would like to leverage the upstream community Cluster API infrastructure providers and reduce the barrier to OpenShift of supporting new platforms.
+
+#### Story 2
+
+As an OpenShift developer, I would like to collaborate with third parties who already have vested interests in maintaining Machine providers for various platforms and environments and benefit from their provider specific expertise as I try to add new features.
+
+#### Story 3
+
+As a cloud developer, I would like to easily onboard new platforms as this process is well documented by CAPI community and any implementation will be able to be leveraged by both Kubernetes and OpenShift customers, increasing the value of implementing a new provider.
+
+#### Story 4
+
+As a developer, I would like to be able to use the same set of tools for infrastructure management in OpenShift as I can for vanilla Kubernetes.
+
+#### Story 5
+
+As a developer, I would like to be able to use infrastructure resource API for managing mixed platform clusters
+
+#### Story 6
+
+As a developer, I would like to have support for hub-spoke clusters. Where management cluster can manage workload clusters that are running on different platforms.
+
+### Implementation Details
+
+First, we need to establish Cluster API resource management by ensuring all required components are successfully installed and running within the OpenShift cluster.
+
+Cluster API will only be present in the cluster (installed by a new operator) if and when a user installs a feature gate.
+We will introduce a new, OpenShift specific, feature gate `ClusterAPIEnabled` and include it within the `TechPreviewNoUpgrade` FeatureSet.
+
+Once installed, this preview will allow the user to create new MachineSets using CAPI and explore the features available within CAPI, for comparison with MAPI.
+
+During this timeframe, any user wanting to migrate or try out the preview will be left to manually migrate the MachineSet, using either the upstream documentation or documentation provider by OpenShift.
+
+This preview will be intended to be used to create day 2 worker MachineSets and is not expected to be integrated into the install process in any way.
+
+### Supported platforms
+
+The techincal preview aims to support: AWS, Azure, GCP.
```

Comment:
not metal3? We have customers that are very keen have virt controlplane and metal3 workers

Reply:
Yes, we would like Baremetal to be one of the supported platfoms early on in the tech preview. Do you see any challenges to that?